### ISSUE TITLE
Rebuild container on docker-compose env file change

### ISSUE THREAD

**Issue by scalp42** [tilt-dev/tilt#5605]
## Describe the Feature You Want

Imagining the current folder structure:

```
my_project:
	- compose
		- variables.env
	- Tiltfile
	- my_project_code
		- Dockerfile
	- docker-compose.yml (with `env_file` being `compose/variables.env`)
```

I'd like Tilt to kick a build when `variables.env` is changed.

### Current Behavior

Nothing happens currently. I've tried `watch_file('./compose')` but it'll only reload the Tiltfile and nothing happens either.

Let me know if I'm missing something and thanks for Tilt 🙌 

EDIT: `listdir('./compose', recursive=True)` seems to do the trick but not sure if that's the right approach.

**Comment by landism** [comment]
Thanks for filing! This is a thing we should fix on the Tilt side, but in the meantime a cleaner workaround might be `docker_compose('docker-compose.yaml', env_file='compose/variables.env')`.

(one key difference being that `listdir` will cause the Tiltfile to be reexecuted, while `env_file` will only cause the docker-compose resources to be updated)

**Comment by landism** [comment]
> (one key difference being that listdir will cause the Tiltfile to be reexecuted, while env_file will only cause the docker-compose resources to be updated)

Upon further investigation, it seems this is theoretically true, but in practice Tilt actually just reexecutes the Tiltfile when the env_file changes.

**Comment by scalp42** [comment]
The issue is that our `docker-compose` reads multiple `env_file` (say 3 services are defined, 3 `env_file` will exist and I'm not sure how to do the mapping in Tilt.

**Comment by scalp42** [comment]
Also related, is there a way also to pass `--no-cache` for `docker_compose` function?

**Comment by landism** [comment]
> The issue is that our `docker-compose` reads multiple `env_file`

Just to confirm, are these specified in your docker-compose.yaml under individual services?
e.g.:
```
version: 3
services:
  myservice:
    env_file:
      - compose/variables.env
```

**Comment by landism** [comment]
The title of this issue says "rebuild container". IIUC, the image build does not get environment variables from the env_file - are you looking for a fresh image build, or just for the container to restart with the new environment variables?

**Comment by nicksieger** [comment]
I have a possible fix in #5750. @scalp42 let me know if the description of the behavior there meets your needs.
